protocols/smtp/src/main/java/org/apache/james/protocols/smtp/core/esmtp/MailSizeEsmtpExtension.java

```diff
@@ -79,8 +82,10 @@ public void destroy() {
     @Override
     public HookResult doMailParameter(SMTPSession session, String paramName,
                                       String paramValue) {
+        MaybeSender tempSender = (MaybeSender) session.getAttachment(SMTPSession.SENDER, State.Transaction);
         return doMailSize(session, paramValue,
-                (String) session.getAttachment(SMTPSession.SENDER, State.Transaction));
+            Optional.ofNullable(tempSender).flatMap(MaybeSender::asOptional)
```

Comment:
what about : 
```
Optional.ofNullable(tempSender).orElseGet(MaybeSender::nullSender).asString()
```

Reply:
This value is only used for logging, and in this class, let's keep the raw `MaybeSender`